docs/modules/mapstore/pages/configuring-a-generic-maploader.adoc

```diff
@@ -0,0 +1,388 @@
+= Using the Generic MapLoader
+:description: With the xref:working-with-external-data.adoc#options[generic MapLoader], you can configure a map to cache data from an external data store. This topic includes an example of how to configure a map with a generic MapLoader that connects to a MySQL database.
+:page-beta: true
+
+{description}
+
+For a list of all supported databases, see the xref:external-data-stores:external-data-stores.adoc#factories[built-in data store factories].
+
+== Before you Begin
+
+You need a xref:external-data-stores:external-data-stores.adoc[data store connection] that's configured on all cluster members.
+
+== Quickstart Configuration
+
+This example shows a basic map configuration for an external datastore called `my-mysql-database`. See xref:data-structures:map.adoc[] for the details of other properties that you include in your map configuration.
+
+[tabs] 
+==== 
+XML:: 
++ 
+-- 
+[source,xml]
+----
+<hazelcast>
+    ...
+    <map name="default">
+        <map-store enabled="true">
+            <class-name>com.hazelcast.mapstore.GenericMapLoader</class-name> <1>
+            <properties>
+                <property name="external-data-store-ref">my-mysql-database</property> <2>
+            </properties>
+        </map-store>
+    </map>
+    ...
+</hazelcast>
+----
+--
+
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  map:
+    default:
+      map-store:
+        enabled: true
+        class-name: com.hazelcast.mapstore.GenericMapLoader <1>
+        properties:
+            external-data-store-ref: my-mysql-database <2>
+----
+--
+Java::
++
+--
+[source,java]
+----
+MapConfig mapConfig = new MapConfig("default");
+MapStoreConfig mapStoreConfig = new MapStoreConfig();
+mapStoreConfig.setClassName("com.hazelcast.mapstore.GenericMapLoader"); <1>
+mapStoreConfig.setProperty("external-data-store-ref", "my-mysql-database"); <2>
+mapConfig.setMapStoreConfig(mapStoreConfig);
+instance().getConfig().addMapConfig(mapConfig);
+----
+--
+====
+
+<1> The xref:configuration-guide.adoc#class-name[class name] of the generic MapLoader that's built into Hazelcast. This implementation reads from the external data store configuration to create a <<mapping, SQL mapping>>.
+<2> The name of your external data store configuration.
+
+[[mapping]]
+== SQL Mapping for the Generic MapStore
+
+When you configure a map with the generic MapLoader, Hazelcast creates a xref:sql:mapping-to-jdbc.adoc[SQL mapping with the JDBC connector]. The name of the mapping is the same name as your map prefixed with `__map-store.`. This mapping is used to read data from the data store or write data to the data store and it is removed whenever the configured map is removed. You can also configure this SQL mapping, using <<external-data-store-ref, configuration properties>>.
```

Comment:
`This mapping is used to read data from the data store or write data to the data store`
for the maploader writing is not valid

Reply:
Removed `or write data to the data store`